core/unit_test/TestGraph.hpp

```diff
@@ -566,8 +566,14 @@ TEST_F(TEST_CATEGORY_FIXTURE(graph), end_of_submit_control_flow) {
 
   Kokkos::parallel_for(
       policy_t(exec_3, 0, 1),
+#if defined(KOKKOS_COMPILER_GNU) && (1010 == KOKKOS_COMPILER_GNU)
+      // Workaround CTAD bug, see 7316.
+// clang-format off
+      FetchValuesAndContribute<data, index_F, 2>(data, {index_D(), index_E()}, index_F, value_F));
```

Comment:
This seems off. Should it be
```suggestion
      FetchValuesAndContribute<decltype(data), index_F, 2>(data, {index_D(), index_E()}, index_F, value_F));
```

Reply:
Yep, thanks @dalg24 for the buggy suggestion (https://github.com/kokkos/kokkos/pull/7323#discussion_r1755285396) :laughing:

Reply:
Any time

Reply:
Lesson learned: I should not apply blindly what you write :wink: